tests/test_api.py

```diff
@@ -275,14 +275,25 @@ def test_api_get_build_one_unauth(testclient):
 
 def test_api_get_build_one_auth(testclient):
     testclient.login()
-    response = testclient.get("api/v1/build/1")
-    response.raise_for_status()
+    success = False
+    for _ in range(50):
+        response = testclient.get("api/v1/build/1")
+        response.raise_for_status()
 
-    r = schema.APIGetBuild.parse_obj(response.json())
-    assert r.status == schema.APIStatus.OK
-    assert r.data.id == 1
-    assert r.data.specification.name == "python-flask-env"
-    assert r.data.status == schema.BuildStatus.COMPLETED.value
+        r = schema.APIGetBuild.parse_obj(response.json())
+        assert r.status == schema.APIStatus.OK
+        assert r.data.id == 1
+        assert r.data.specification.name == "python-flask-env"
+        if r.data.status in [
+            schema.BuildStatus.QUEUED.value,
+            schema.BuildStatus.BUILDING.value,
+        ]:
+            time.sleep(10)
+            continue
+        assert r.data.status == schema.BuildStatus.COMPLETED.value
+        success = True
+        break
+    assert success
```

Comment:
Okay, let's hope we don't need 50 retries here. A slightly cleaner way of doing this is to just `return` when the build is completed rather than the success/break/assert:

```python
def test_api_get_build_one_auth(testclient):
    testclient.login()
    for _ in range(50):
        response = testclient.get("api/v1/build/1")
        response.raise_for_status()
                r = schema.APIGetBuild.parse_obj(response.json())
        assert r.status == schema.APIStatus.OK
        assert r.data.id == 1
        assert r.data.specification.name == "python-flask-env"
        if r.data.status in [
            schema.BuildStatus.QUEUED.value,
            schema.BuildStatus.BUILDING.value,
        ]:
            time.sleep(10)
            continue
        assert r.data.status == schema.BuildStatus.COMPLETED.value
        return
```

but it's up to you if you think it worth the change.

Reply:
The suggested change is problematic: if the build is stuck in the queued/building state for 50 iterations, the loop will exit silently and the test will succeed. I want to detect this as well, which is why I check for success outside of the loop.